src/internal/SetupResumeAcceptor.cpp

```diff
@@ -143,8 +143,19 @@ void SetupResumeAcceptor::processFrame(
         break;
       }
 
-      removeConnection(transport);
-      onResume(std::move(transport), std::move(resumeParams));
+      if (!onResume(transport, std::move(resumeParams))) {
+        transport->outputFrameOrEnqueue(frameSerializer->serializeOut(
+            Frame_ERROR::connectionError("Resumption Failure")));
+        eventBase_->runInEventBaseThread(
```

Comment:
(Post IRL conversation) This definitely deserves a comment, because to the untrained eye (mine) it looks like an unnecessary use of `runInEventBaseThread()`.